menas/src/main/scala/za/co/absa/enceladus/menas/services/VersionedModelService.scala

```diff
@@ -139,11 +146,15 @@ abstract class VersionedModelService[C <: VersionedModel with Product with Audit
 
   private[services] def updateFuture(username: String, itemName: String, itemVersion: Int)(transform: C => Future[C]): Future[Option[C]] = {
     for {
-      version <- getVersion(itemName, itemVersion)
-      transformed <- if (version.isEmpty) {
+      versionToUpdate <- getVersion(itemName, itemVersion)
+      latestVersion <- getLatestVersionNumber(itemName)
```

Comment:
The original idea for this function was to perform updates based on any version, so `getVersion` was used. Now that we are enforcing updates only based on the latest version, we can change the `getVersion` call to `getLatestVersion` and skip the call to `getLatestVersionNumber`. 

Since we get the latest version's number from `getLatestVersion` we would just need to compare it against the `itemVersion` parameter, to make sure that the user is actually updating the intended entity version. 

(e.g. **A** starts updating when the latest version is 1. **B** updates the entity to version 2 before **A** submits. **A** will send the update request with `itemVersion = 1`, the call to `getLatestVersion` will return an entity with latest version 2, and **A** will see that their update is outdated.)